### PR TITLE
chore: upgrade to azure-sdk v41.0.0

### PR DESCRIPTION
**Pull request by andyzhangx**
**Reason for Change**:

This PR upgrade azure-sdk to support features, e.g. BYOK encryption on host

```
vi go.mod

grep "2019-07-01/compute" ./* -R

sed -i 's/2019-07-01\/compute/2019-12-01\/compute/g' ./pkg/armhelpers/*.go
sed -i 's/2019-07-01\/compute/2019-12-01\/compute/g' ./pkg/engine/*.go
sed -i 's/2019-07-01\/compute/2019-12-01\/compute/g' ./pkg/operations/kubernetesupgrade/*.go
sed -i 's/2019-07-01\/compute/2019-12-01\/compute/g' ./pkg/armhelpers/azurestack/*.go
sed -i 's/2019-07-01\/compute/2019-12-01\/compute/g' ./pkg/armhelpers/utils/*.go
sed -i 's/2019-07-01\/compute/2019-12-01\/compute/g' ./pkg/engine/transform/transform.go
sed -i 's/2019-07-01\/compute/2019-12-01\/compute/g' ./cmd/rotate_certs_test.go

go mod vendor
go mod tidy
```

**Issue Fixed**:

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: pkg/armhelpers/httpMockClient.go [hunk 1]
```diff
@@ -19,7 +19,7 @@ const (
 	tenantID                                   = "19590a3f-b1af-4e6b-8f63-f917cbf40711"
 	resourceGroup                              = "TestResourceGroup"
 	computeAPIVersion                          = "2019-07-01"
-	diskAPIVersion                             = "2019-07-01"
+	diskAPIVersion                             = "2019-11-01"
```
Comment (mboersma) [inline pkg/armhelpers/httpMockClient.go:22]: Is this related to the compute API bump?

**Comment by CecileRobertMichon** [comment]
why not move to 0.41.0?

**Comment by zqingqing1** [comment]
@andyzhangx is that good to bump it a little higher to 41.0.0?

**Comment by andyzhangx** [comment]
/hold
let me bump to `41.0.0`

File: pkg/armhelpers/httpMockClient.go (hunk 1 above)
Comment (andyzhangx) [reply pkg/armhelpers/httpMockClient.go:22]: correct, the disk api version is already `2019-11-01` in v40.2.0, need to change the version in mock client, otherwise unit test failed.

**Comment by andyzhangx** [comment]
/hold cancel
already bumped to `v41.0.0`

**Comment by andyzhangx** [comment]
@zqingqing1 this PR is merged in master branch, are you going to make it into `aks-release-v0.47.0-1` branch?
